File: lib/src/core/domain/padels/entities/padel.group.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'padel.dart';

part 'padel.group.freezed.dart';
part 'padel.group.g.dart';

@freezed
class PadelGroupModel with _$PadelGroupModel {
  factory PadelGroupModel({
    @Default(-1) int id,
    @Default('') String name,
    @Default('') String icon,
    String? color,
    @Default(1) int sequence,
    @Default(true) bool enabled,
    DateTime? createdAt,
    DateTime? updatedAt,
    List<PadelModel>? Padels,
  }) = _PadelGroupModel;
  factory PadelGroupModel.fromJson(Map<String, dynamic> json) =>
      _$PadelGroupModelFromJson(json);
}

File: src/modules/padels/infrastructure/dto/padel.group.ts
import { PadelAddDto } from './padel.add.dto';
import { IsNotEmpty } from 'class-validator';

export class PadelGroupDto {
  readonly id: number;

  @IsNotEmpty({ message: 'Name is required!' })
  readonly name: string;

  readonly icon: string;

  readonly color: string;

  readonly sequence: number;

  readonly enabled: boolean;

  readonly Padel: PadelAddDto[];
}
